### PR TITLE
Refactoring of the utils

### PR DESCRIPTION
**Pull request by Remit**
[TICKET](https://hivemq.kanbanize.com/ctrl_board/42/cards/11323/details/)

**Motivation**

- removing various kinds of utils from the HiveMQ code base to reduce the complexity and increase maintainability
- restructuring some classes to better use the OOP paradigm
- renaming and restructuring some unit tests to better reflect the test cases
- fixing some test cases
- adding null pointer checks

**Changes**

Removed utils:

- ChannelAttributes
- ChannelUtils
- ClientSessions
- DefaultSslEngineUtil
- MacAddressFormatter
- PhysicalMemoryUtil
- PublishUtil
- ResourceUtils
- TemporaryFileUtils
- UcaughtExceptionHandler
- Validators

### CODE REVIEW

File: src/main/java/com/hivemq/util/ThreadFactoryUtil.java [hunk 1]
```diff
@@ -18,12 +18,11 @@
 
 import com.google.common.util.concurrent.ThreadFactoryBuilder;
 import com.hivemq.extension.sdk.api.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.util.concurrent.ThreadFactory;
 
-/**
- * @author Florian Limpöck
```
Comment (hlohse) [inline src/main/java/com/hivemq/util/ThreadFactoryUtil.java:25]: We're keeping author tags in the CE, right?

File: src/test/java/com/hivemq/diagnostic/data/NetworkInterfaceInformationTest.java [hunk 2]
```diff
@@ -17,16 +17,11 @@
...
 import org.junit.Test;
-import org.mockito.MockitoAnnotations;
 
 import java.net.NetworkInterface;
 
-import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.*;
 
-/**
- * @author Florian Limpöck
```
Comment (hlohse) [inline src/test/java/com/hivemq/diagnostic/data/NetworkInterfaceInformationTest.java:27]: We're keeping author tags in the CE, right?

File: src/main/java/com/hivemq/diagnostic/data/NetworkInterfaceInformation.java [hunk 3]
```diff
@@ -26,11 +25,13 @@
 import java.util.Collections;
 import java.util.Enumeration;
 
-/**
- * @author Dominik Obermaier
```
Comment (hlohse) [inline src/main/java/com/hivemq/diagnostic/data/NetworkInterfaceInformation.java:30]: We're keeping author tags in the CE, right?

**Review by hlohse — APPROVED** [review]
LGTM but please revert the removed author tags. We're keeping them in the CE, right? (I stopped commenting after the first three I noticed, but more have been removed.)

File: src/main/java/com/hivemq/util/ThreadFactoryUtil.java (hunk 1 above)
Comment (Remit) [reply src/main/java/com/hivemq/util/ThreadFactoryUtil.java:25]: From what I know from Silvio, we keep authors only for APIs

**Review by LBrandl — APPROVED** [review]

**Pull request merged**